### PR TITLE
Adding pinned post to DCR model

### PR DESCRIPTION
**Pull request by alinaboghiu**
## What does this change?

Adding an optional pinnedPost field to DotcomRenderingDataModel. 

When creating the DCR json for liveblogs we should include the entire pinned post explicitly. We need to do this because we always want to display the pinned post on the first page, but the pinned post may not always exist amongst the blocks on the first page, so the block id isn't enough. 

The field is optional seeing as not all blogs have a pinned post.

### CODE REVIEW

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 1]
```diff
@@ -175,14 +177,15 @@ object DotcomRenderingDataModel {
...
+      page = page,
+      request = request,
+      pagination = None,
       linkedData = Nil, // TODO
       mainBlock = blocks.main,
       bodyBlocks = blocks.body.getOrElse(Nil),
-      pageType,
-      page.related.hasStoryPackage,
+      pageType = pageType,
+      hasStoryPackage = page.related.hasStoryPackage,
```
Comment (oliverlloyd) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:187]: I assume these changes are either boyscouting or just the IDE having ideas?

**Review by joecowton1 — APPROVED** [review]
LGTM

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 2]
```diff
@@ -328,23 +340,27 @@ object DotcomRenderingDataModel {
...
       .flatMap(entry => entry._2.asOpt[String])
 
+    def toDCRBlock = { block: APIBlock =>
+      Block(block, page, shouldAddAffiliateLinks, request, false, calloutsUrl, contentDateTimes)
+    }
+
     val mainMediaElements =
       mainBlock
-        .map(block => Block(block, page, shouldAddAffiliateLinks, request, true, calloutsUrl, contentDateTimes))
+        .map(toDCRBlock)
```
Comment (oliverlloyd) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:352]: In the `toDCRBlock` abstraction above we're now hardcoding one of the parameters to `false` but in the original code it was true for `mainBlock` and only false for `bodyBlocks`. Just saying.

**Review by oliverlloyd — APPROVED** [review]
Looks great (to me). I noticed a potential unwanted side effect to an abstraction but I'm happy for you to fix or ignore as appropriate

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala (hunk 1 above)
Comment (alinaboghiu) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:187]: the former yes

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala (hunk 2 above)
Comment (alinaboghiu) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:352]: good spot, I'll fix

**Pull request merged**